### DISCUSSION TITLE
How to programmatically set the focus?

### DISCUSSION THREAD

**Discussion by eugenevk** [quasarframework/quasar#12206] [General - Components / Directives / etc] [answered]
How can I programmatically set the focus on a QInput field, e.g. after validation?

**Comment by yusufkandemir** [comment, marked as answer]
You can use the `focus()` method.

![image](https://user-images.githubusercontent.com/6266078/151251526-59fa5d85-0d14-42fe-a788-fbee640f7613.png)
https://quasar.dev/vue-components/input#qinput-api


Example:
```html
<template>
  <q-btn label="Focus to input" @click="focusToInput" />

  <q-input v-model="model" ref="inputRef" />
</template>

<script setup>
import { ref } from 'vue';

const model = ref('');
const inputRef = ref();

function focusToInput () {
  inputRef.value.focus();
}
</script>
```


Judging from `e.g. after validation?`, I guess you want to focus on the first field with a validation error. If that is your aim, [QForm](https://quasar.dev/vue-components/form) got you covered. Check out [this example](https://quasar.dev/vue-components/form#example--basic).

**Reply by eugenevk** [reply]
Thanks @yusufkandemir I saw the form example, but it doesn't work on my QDialog (see my other discussion topic #12205 ).

**Reply by hhosseinipaymentshield** [reply]
Did you manage to get this to work? I have the same problem.

**Reply by eugenevk** [reply]
Hi @hhosseinipaymentshield, I have resolved it like this (for a custom component called InputText using QInput):
```
<template>
  <q-input
    :autofocus="focus"
    :class="[extraClass, width]"
    hide-bottom-space
    :label="label"
    label-color="primary"
    :label-slot="required"
    :lazy-rules="required"
    :model-value="model"
    outlined
    :readonly="readonly"
    :rules="[required ? isRequired : null]"
    stack-label
    @update:modelValue="$emit('update:modelValue', $event)"
  >
    <template v-if="required" #label>
      {{ label }} <span class="text-weight-bold text-red">*</span>
    </template>
  </q-input>
</template>

<script setup>
import { defineProps, defineEmits } from 'vue'

import validationRules from '../composables/validationRules'

defineEmits(['update:modelValue'])
defineProps(['extraClass', 'focus', 'label', 'model', 'readonly', 'required', 'width'])

const { isRequired } = validationRules
</script>
```
And in another component I use this custom component as follows:
```
<InputText :label="'Name'" v-model="name" :width="'col-4'" :focus="true" />
```
It automatically sets the focus correctly. It also sets the focus automatically on the field if the validation rules are not satisfied.
Hope this helps.
